winpr/libwinpr/ncrypt/ncrypt_pkcs11.c

```diff
@@ -543,7 +543,7 @@ static BOOL convertKeyType(CK_KEY_TYPE k, LPWSTR dest, DWORD len, DWORD* outlen)
 	}
 	else
 	{
-		if (retLen + 1 > len)
+		if (len && retLen + 1 > len)
```

Comment:
Looks buggy:
1. you skip this if `len == 0`
2. a few lines below a `memcpy()` with a (unchecked) `retLen >= 0`?

Reply:
@hardening I think the check `if (len > 0)` should be part of the else above.

Reply:
reworked that.